com.sap.cloud.lm.sl.cf.process/src/test/java/com/sap/cloud/lm/sl/cf/process/steps/DetermineDesiredStateAchievingActionsStepTest.java

```diff
@@ -124,16 +133,25 @@
     private final boolean hasServicesPropertiesChanged;
     private final boolean hasUserPropertiesChanged;
     private final Set<ApplicationStateAction> expectedAppStateActions;
-
+    private final List<CloudBuild> cloudBuilds;
     private RestartParameters appRestartParameters;
 
     @Mock
     private ApplicationStartupStateCalculator appStateCalculator;
 
+    @Mock
+    private CloudControllerClient client;
+
+    @Mock
+    private CloudApplicationExtended cloudApplicationExtended;
```

Comment:
Usually, the mocks we create via the `@Mock` annotation are objects with behavior (clients, factories, etc.) and not POJOs. POJOs we typically mock wherever necessary directly with `Mockito.mock()`. That's because objects with behavior are more likely to get reused in multiple methods in the test, while POJOs are usually used in just one place and it's always a good idea to [limit the scope of a variable as much as possible](https://softwareengineering.stackexchange.com/questions/307346/why-is-it-good-programming-practice-to-limit-scope).
Additionally, mocks with `@Mock` are used whenever they are injected into the tested class via `@InjectMocks` (see the parent class of this test).

Reply:
Thanks for the info, done